File: Stock-Trader-App/Models/TradeIdea.cs
using Stock_Trader_App.Models.Enums;

namespace Stock_Trader_App.Models
{
    public class TradeIdea
    {
        public int Id { get; set; }

        public string Ticker { get; set; }

        public double StrikePrice { get; set; }
        public DateTime ExpirationDate { get; set; }
        public double? Delta { get; set; }
        public double Premium { get; set; }
        public double IV { get; set; }
        public int DTE { get; set; }
        public double ROI { get; set; }
        public string OptionType { get; set; } 

        public Strategy Strategy { get; set; }
        public string Notes { get; set; }
        public Status Status { get; set; } = Status.Idea;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Front-end/stock-trader-ui/src/models/TradeIdea.ts
export type TradeIdea = {
  id: number;
  ticker: string;

  option: {
    type: number;
    strike: number;
    expiration: string;
    premium: number;
    iv: number;
    dte: number;
    roi: number;
    delta: number | null;
  };

  strategy: "CoveredCall" | "CashSecuredPut" | "Other";
  notes: string;
  status: "Idea" | "Opened" | "Closed";
};
